### PR TITLE
feat: allow getting health check status via control API

### LINKED ISSUE #2798: [DISCUSS] provide a control interface to expose APISIX internal state
We can let APISIX listen a specific port and provide some APIs just like the current admin server.

Those APIs include:
* /healthcheck: show the status of healthcheck. See #2797.

### PR DESCRIPTION
**Pull request by spacewander**
Fix #2798

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/control/v1.lua [hunk 1]
```diff
@@ -49,11 +54,62 @@ function _M.schema()
...
 return {
     -- /v1/schema
     {
         methods = {"GET"},
         uris = {"/schema"},
         handler = _M.schema,
+    },
+    -- /v1/healthcheck
+    {
+        methods = {"GET"},
```
Comment (tokers) [inline apisix/control/v1.lua:111]: Just a proposal the output maybe too large if number of upstreams are large, while sometimes users will want to focus on a single upstream, so API `/v1/healthcheck/{id}` maybe useful.

Comment (Firstsawyou) [reply apisix/control/v1.lua:111]: I think it is very useful to support the two methods `/v1/healthcheck/` and `/v1/healthcheck/{id}`.

Comment (spacewander) [reply apisix/control/v1.lua:165]: @tokers @Firstsawyou 
Added.

File: t/plugin/proxy-rewrite.t [hunk 2]
```diff
@@ -297,6 +297,7 @@ passed
 
 
 === TEST 9: rewrite host + scheme
+--- LAST
```
Comment (tokers) [inline t/plugin/proxy-rewrite.t:300]: Is this for debuging?

Comment (spacewander) [reply t/plugin/proxy-rewrite.t]: Fixed

**Review by tokers — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Comment by gzhsnail** [comment]
Is it possible to view the results of health examination now?
请问现在是已经可以查看到健康检查的结果了吗？

**Comment by Firstsawyou** [comment]
> Is it possible to view the results of health examination now?
> 请问现在是已经可以查看到健康检查的结果了吗？

Please look here: https://github.com/apache/apisix/blob/master/doc/control-api.md#get-v1healthcheck

**Pull request merged**